sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/ServiceSASSignatureValues.java

```diff
@@ -232,20 +234,18 @@ public ServiceSASSignatureValues canonicalName(String canonicalName) {
     /**
      * The canonical name of the object the SAS user may access.
      *
-     * @throws RuntimeException If urlString is a malformed URL.
+     * @throws IllegalArgumentException If urlString is a malformed URL.
      */
     public ServiceSASSignatureValues canonicalName(String urlString, String accountName) {
-        URL url = null;
         try {
-            url = new URL(urlString);
+            URL url = new URL(urlString);
+            StringBuilder canonicalName = new StringBuilder("/blob");
+            canonicalName.append('/').append(accountName).append(url.getPath());
+            this.canonicalName = canonicalName.toString();
         } catch (MalformedURLException e) {
-            throw new RuntimeException(e);
+            logger.logAndThrow(new IllegalArgumentException(e));
         }
 
-        StringBuilder canonicalName = new StringBuilder("/blob");
```

Comment:
The statements here does not have the chance of throwing MalformatedURLException. It is better to leave outside.